ephios/core/notifications/types.py

```diff
@@ -0,0 +1,340 @@
+
+
+class ProfileUpdateNotification(AbstractNotificationHandler):
+    slug = "ephios_profile_update"
+    title = _("Your profile has been edited")
+
+    @classmethod
+    def send(cls, user: UserProfile, **kwargs):
+        return Notification.objects.create(slug=cls.slug, user=user, data=kwargs)
+
+    @classmethod
+    def get_subject(cls, notification):
+        return _("ephios account updated")
+
+    @classmethod
+    def as_plaintext(cls, notification):
+        return _(
+            "You're receiving this email because your account at ephios has been updated.\n"
+            "You can see the changes in your profile: {url}\n"
+            "Your username is your email address: {email}\n"
+        ).format(
+            url=urljoin(settings.SITE_URL, reverse("core:profile")), email=notification.user.email
+        )
+
+
+class NewProfileNotification(AbstractNotificationHandler):
+    slug = "ephios_new_profile"
+    title = _("A new profile has been created")
+    unsubscribe_allowed = False
+
+    @classmethod
+    def send(cls, user: UserProfile, **kwargs):
+        uid = urlsafe_base64_encode(force_bytes(user.id))
+        token = default_token_generator.make_token(user)
+        return Notification.objects.create(
+            slug=cls.slug, user=user, data={**{"uidb64": uid, "token": token}, **kwargs}
+        )
+
+    @classmethod
+    def get_subject(cls, notification):
+        return _("Welcome to ephios!")
+
+    @classmethod
+    def as_plaintext(cls, notification):
+        reset_link = reverse("password_reset_confirm", kwargs=notification.data)
+        return _(
+            "You're receiving this email because a new account has been created for you at ephios.\n"
+            "Please go to the following page and choose a password: {url}\n"
+            "Your username is your email address: {email}\n"
+        ).format(url=urljoin(settings.SITE_URL, reset_link), email=notification.user.email)
+
+    @classmethod
+    def as_html(cls, notification):
+        return render_to_string(
+            "core/new_account_email.html",
+            {
+                **{"site_url": settings.SITE_URL, "email": notification.user.email},
+                **notification.data,
+            },
+        )
+
+
+class NewEventNotification(AbstractNotificationHandler):
+    slug = "ephios_new_event"
+    title = _("A new event has been added")
+
+    @classmethod
+    def send(cls, event: Event, **kwargs):
+        notifications = []
+        for user in get_users_with_perms(event, only_with_perms_in=["view_event"]):
+            notifications.append(
+                Notification(slug=cls.slug, user=user, data=dict(event_id=event.id, **kwargs))
+            )
+        Notification.objects.bulk_create(notifications)
+
+    @classmethod
+    def get_subject(cls, notification):
+        event = Event.objects.get(pk=notification.data.get("event_id"))
+        return _("New {type}: {title}").format(type=event.type, title=event.title)
```

Comment:
Translation here will be tricky, as discussed a few months ago.
Instead of "Neuer Dienst", "Neuer Ausbildung", "Neuer Biwak" we will have to e.g. drop the "new", prepend a "Rückmeldung erwünscht", or append an "added", like "Biwak hinuzgefügt: ...".